crates/e2e-test-utils/src/testsuite/assertions.rs

```diff
@@ -0,0 +1,229 @@
+//! Assertions that can be checked in tests.
+
+use crate::testsuite::{BoxFuture, ResultStore};
+use alloy_primitives::{BlockNumber, B256, U256};
+use eyre::Result;
+
+/// An assertion that can be checked against stored results
+pub trait Assertion: Send + 'static {
+    /// Assert a condition on the stored results
+    fn assert<'a>(&'a mut self, results: &'a ResultStore) -> BoxFuture<'a, Result<()>>;
+}
```

Comment:
this is more or less similar to an action because in order to assert something we need access to the environment itself, what I'm thinking is something like.

```
let action = MineBlockAction {};
let assert_block = AssertRpcBlock {};
runner.execute(async move|env| {
  action.execute(env).await?;
  assert_block.assert(env).await?;
})
```

Reply:
yep, as we talked online, now assertions and actions are unified, actions can make the required assertions internally if needed, a test suite is just a set of actions